### PR TITLE
cronjob: Add CRUD tests for CronJob API verbs

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind feature
/sig apps
/area test

#### What this PR does / why we need it:
To be able to promote cronjobs to GA we need test ensuring all operations are tested.
Ref https://github.com/kubernetes/kubernetes/pull/98965
Ref https://apisnoop.cncf.io/1.21.0/beta/batch

#### Special notes for your reviewer:
/assign @damemi 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/19-Graduate-CronJob-to-Stable 
```

### CODE REVIEW

File: test/e2e/apps/cronjob.go [hunk 1]
```diff
@@ -280,6 +286,140 @@ var _ = SIGDescribe("CronJob", func() {
...
+		cjStatus := batchv1beta1.CronJobStatus{
+			LastScheduleTime: &now1,
+		}
+		cjStatusJSON, err := json.Marshal(cjStatus)
+		framework.ExpectNoError(err)
+		patchedStatus, err := cjClient.Patch(context.TODO(), createdCronJob.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patchedstatus":"true"}},"status":`+string(cjStatusJSON)+`}`),
+			metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedStatus.Status.LastScheduleTime, &now1, "patched object should have the applied lastScheduleTime status")
```
Comment (damemi) [inline test/e2e/apps/cronjob.go:387]: This fails CI
```
test/e2e/apps/cronjob.go:289
Feb 22 18:38:32.590: patched object should have the applied lastScheduleTime status
Expected
    <*v1.Time | 0xc0026467e0>: {
        Time: 2021-02-22T18:38:32Z,
    }
to equal
    <*v1.Time | 0xc0003ce640>: {
        Time: 2021-02-22T18:38:32Z,
    }
test/e2e/apps/cronjob.go:387
```
looks like the object reference is throwing it off here? Maybe looking for a semantic equal

**Review by damemi — CHANGES_REQUESTED** [review]

File: test/e2e/apps/cronjob.go (hunk 1 above)
Comment (soltysh) [reply test/e2e/apps/cronjob.go]: Switched to `metav1.Time.Equal` which should be better.

**Comment by soltysh** [comment]
/retest
Test failures are unrelated to this PR.

**Pull request merged**